src/julia.h

```diff
@@ -1446,9 +1465,46 @@ JL_DLLEXPORT JL_CONST_FUNC jl_tls_states_t *(jl_get_ptls_states)(void);
 #ifndef JULIA_ENABLE_THREADING
 extern JL_DLLEXPORT jl_tls_states_t jl_tls_states;
 #define jl_get_ptls_states() (&jl_tls_states)
+STATIC_INLINE int8_t jl_gc_state_set_and_save_(int8_t state, int8_t old_state)
+{
+    (void)state;
+    return old_state;
+}
+#define jl_gc_state_set_and_save__(state, old_state, ...)       \
+    jl_gc_state_set_and_save_(state, old_state)
+// Add two arguments, one for the default value of `old_state`, one to satisfy
+// ISO C++11 standard.
+#define jl_gc_state_set_and_save(...)           \
+    jl_gc_state_set_and_save__(__VA_ARGS__, 0, 0)
```

Comment:
I think the only question is what does MSVC support and IIRC it supports the `__VA_ARGS__` version in c++ mode. @tkelman